src/twisted/mail/interfaces.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+Interfaces for L{twisted.mail}.
+
+@since: 16.5
+"""
+
+from __future__ import absolute_import, division
+
+from zope.interface import Interface
+
+
+class IClientAuthentication(Interface):
+
+    def getName():
+        """
+        Return an identifier associated with this authentication scheme.
+
+        @rtype: C{bytes}
+        """
+
+    def challengeResponse(secret, challenge):
+        """
+        Generate a challenge response string.
+        """
+
+
+
+class IMessageDelivery(Interface):
+
+    def receivedHeader(helo, origin, recipients):
+        """
+        Generate the Received header for a message
+
+        @type helo: C{(str, str)}
+        @param helo: The argument to the HELO command and the client's IP
+        address.
+
+        @type origin: C{Address}
+        @param origin: The address the message is from
+
+        @type recipients: C{list} of L{User}
+        @param recipients: A list of the addresses for which this message
+        is bound.
+
+        @rtype: C{str}
+        @return: The full \"Received\" header string.
+        """
+
+    def validateTo(user):
```

Comment:
You should take a pass through the doc string,
and change C{str} -> L{str},   C{list} -> L{list}
C{Deferred} -> L{Deferred}

etc.